src/pudl/transform/sec10k.py

```diff
@@ -493,10 +695,100 @@ def core_sec10k__assn_sec10k_filers_and_eia_utilities(
     companies are not utilities, and most utilities are still unmatched.
     """
     sec_eia_assn = (
-        core_sec10k__parents_and_subsidiaries.loc[
+        raw_sec10k__parents_and_subsidiaries.loc[
             :, ["central_index_key", "utility_id_eia"]
         ]
         .dropna()
         .drop_duplicates(subset=["central_index_key", "utility_id_eia"])
     )
     return sec_eia_assn
+
+
+@dg.asset(io_manager_key="pudl_io_manager", group_name="core_sec10k")
+def core_sec10k__assn_exhibit_21_subsidiaries_and_filers(
+    core_sec10k__quarterly_filings,
+    core_sec10k__quarterly_company_information,
+    core_sec10k__quarterly_exhibit_21_company_ownership,
+) -> pd.DataFrame:
+    """Match Ex. 21 subsidiaries to SEC 10k filing companies.
+
+    Create an association between ``subsidiary_company_id_sec10k``
+    and ``central_index_key``.
+    """
+    filer_info_df = core_sec10k__quarterly_company_information.merge(
+        core_sec10k__quarterly_filings[["filename_sec10k", "report_date"]],
+        how="left",
+        on="filename_sec10k",
+    )
```

Comment:
I only chose to do this merge because I wanted `report_date` in the company information table but didn't want to use the `out_sec10k__quarterly_company_information` because I thought it was better to keep output tables downstream of core tables. Also, I thought it might make sense to merge the matched `subsidiary_company_id` onto `out_sec10k__quarterly_company_information` for filers that are also subs. Thoughts?